cmd/boulder-ra/main.go

```diff
@@ -161,8 +161,14 @@ func main() {
 	cmd.FailOnError(err, "Failed to load credentials and create gRPC connection to Publisher")
 	pubc = bgrpc.NewPublisherClientWrapper(pubPB.NewPublisherClient(conn))
 
+	if (len(c.RA.CTLogGroups2) == 0) {
+		cmd.FailOnError(fmt.Errorf("CTLogGroups2 needs to be provided."), "")
+	}
 	for _, g := range c.RA.CTLogGroups2 {
 		for _, l := range g.Logs {
+			if (g.Logs == nil || len(g.Logs) == 0) {
```

Comment:
I don't think this check works as intended because of its location inside of the `for ... range g.Logs` scope.  If `g.Logs` is `nil` or empty then it won't be iterated by `range` and the `if` statement in the body won't be evaluated.

I think what you want is to hoist the check up one scope into the outer `for` loop over `c.RA.CTLogGroups2`. You can also simplify the `if` clause because `len(nil) == 0`. 
```
for _, g := range c.RA.CTLogGroups2 {
  if len(g.Logs) == 0 {
     ...
  }
  for _, l := range g.Logs {
     ...
  }
  ...
}
```